lib/reverse_sentence.rb

```diff
@@ -1,4 +1,48 @@
 # A method to reverse the words in a sentence, in place.
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+  return my_sentence if (my_sentence.length == 0 || my_sentence.length == 1)
+
+  #first reverse each word in place
+
+  i = 0
+  last = my_sentence.length - 1
+
+  while i < my_sentence.length
+
+      start_of_word = i
+      end_of_word = i
+
+      until my_sentence[i + 1] == " " || end_of_word == last
+        end_of_word += 1
+        i += 1
+      end
+
+      temp = ""
+      step = start_of_word
+      backstep = end_of_word
+      while step <= (start_of_word + end_of_word) / 2
```

Comment:
To avoid repeated code, consider creating a method that takes three parameters: my_sentence, start_of_word and end_of_word. The method than reverses the characters in my_sentence beginning at the index start_of_word until and including character at index end_of_word. 
You could call this catered reverse method again at line 38, passing in start_of_word as 0 and end_of_word as my_sentence.length-1.